templates/statefulset.yaml

```diff
@@ -53,14 +53,22 @@ spec:
       {{- end }}
       containers:
         - env:
-            - name: CONFIG_PATH
-              value: /etc/config/infinispan.yaml
             - name: JAVA_OPTIONS
               value: {{ .Values.container.extraJvmOpts }}
-            - name: IDENTITIES_PATH
-              value: /etc/security/user/identities.yaml
           image: {{ .Values.image }}
           imagePullPolicy: IfNotPresent
+          command:
+            - /bin/sh
+          args:
+            - -c
+            - >-
+              /opt/infinispan/bin/cli.sh --file /etc/security/user/identities-batch &&
```

Comment:
This is used to populate the `users.properties` and `group.properties` files used by the server. Populating the files this way means that credentials are not stored in plain-text.

I have kept the the `identities.yaml` format as it makes it easy for users to provide the config via `--set-file=-set-file security.identities=example-identities.yaml` and mirrors the format currently used by the operator.

Reply:
This doesn't really work with `security.secretName` as the user would actually have to provide a secret with the `infinispan-batch` file.

Maybe we should move away from the `idenitites.yaml` format after all and just allow users to provide the `infinispan-batch` content via a Secret or `.Values.security.batch` string. 

IMO utilising the CLI syntax is better than the custom yaml format as it's inline with the rest of the Infinispan ecosystem. As we move the operator away from the config-generator, we can also consider using CLI syntax there.

WDYT?

Reply:
Yeah, very good option. CLI more flexible way than fixed yaml format we need to change every time we need add additional option.
For monitoring, we could generate a static username/password as operator does for admin endpoint and utilize 11223 port for it.

Reply:
> For monitoring, we could generate a static username/password as operator does for admin endpoint and utilize 11223 port for it.

+1. I was trying to avoid this, however we may have todo this as prometheus only supports BASIC authentication this is not enabled on the user endpoint.

I'll add these changes tomorrow.

Reply:
Added as a new commit. Currently we don't automatically create the ServiceMonitor spec, do you know if it's possible with helm charts to determine if a type is available before creating the resource?

Reply:
> Added as a new commit. Currently we don't automatically create the ServiceMonitor spec, do you know if it's possible with helm charts to determine if a type is available before creating the resource?

Yes, sure. Please see the Router creation code template

Reply:
@dmvolod I have added the logic to create the ServiceMonitor, however it's not being detected by `openshift-user-workload-monitoring` (operator ServiceMonitor in another namespace is detected) ... have I missed anything obvious?

Reply:
It's strange as this and infinispan-operator code just validates ServiceMonitor CRD availability